dubbo-rpc/dubbo-rpc-dubbo/src/test/java/org/apache/dubbo/rpc/protocol/dubbo/telnet/InvokerTelnetHandlerTest.java

```diff
@@ -68,6 +69,45 @@ public void testInvokeDefaultSService() throws RemotingException {
         assertTrue(result.contains("Use default service org.apache.dubbo.rpc.protocol.dubbo.support.DemoService.\r\n\"ok\"\r\n"));
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testInvokeByPassingNullValue() throws RemotingException {
+        mockInvoker = mock(Invoker.class);
+        given(mockInvoker.getInterface()).willReturn(DemoService.class);
+        given(mockInvoker.getUrl()).willReturn(URL.valueOf("dubbo://127.0.0.1:20883/demo"));
+        given(mockInvoker.invoke(any(Invocation.class))).willReturn(new RpcResult("ok"));
+        mockChannel = mock(Channel.class);
+        given(mockChannel.getAttribute("telnet.service")).willReturn("org.apache.dubbo.rpc.protocol.dubbo.support.DemoService");
+        given(mockChannel.getLocalAddress()).willReturn(NetUtils.toAddress("127.0.0.1:5555"));
+        given(mockChannel.getRemoteAddress()).willReturn(NetUtils.toAddress("127.0.0.1:20883"));
+
+        DubboProtocol.getDubboProtocol().export(mockInvoker);
+
+        // pass null value to parameter of primitive type
+        try {
+            invoke.telnet(mockChannel, "DemoService.add(null, 2)");
+            fail("It should cause a NullPointerException here.");
+        } catch (NullPointerException ex) {
+            String message = ex.getMessage();
+            assertEquals("The type of No.1 parameter is primitive(int), but the value passed is null.", message);
+        }
+
+        try {
+            invoke.telnet(mockChannel, "DemoService.add(1, null)");
+            fail("It should cause a NullPointerException here.");
```

Comment:
Is the message right ?  "It shouldn't cause a NullPointerException here."

Reply:
The message is right, because we expect  the code `invoke.telnet(mockChannel, "DemoService.add(1, null)")` to throw a NullPointerException. I think it was the word "here" made you confused, so I changed the word "here" to "by the above code" now. Please review it again.